### PR TITLE
Kevin/post todo

### PR DESCRIPTION
**Pull request by kevinrosarioc**
Create method in todo/controller is implemented, and tested in todo.controller.js. 
Please note that I had to change the initialization of the data base object from using const and require to import from because it didn't work the other way around.
I also updated api.yml to provide better information about the post method.

### CODE REVIEW

**Review by Albertucho22 — DISMISSED** [review]

File: backend/test/todo.controller.js [hunk 1]
```diff
@@ -21,3 +21,34 @@ describe("Stub test for controller and server", () => {
           .equal("test");
       }));
 });
+
+describe("test cases for create controller method", () => {
+  it("should add a new todo when body has all properties", () => {
```
Comment (fernandogruning) [inline backend/test/todo.controller.js]: Please add test cases for empty bodies

File: backend/test/todo.controller.js [hunk 2]
```diff
@@ -21,3 +21,34 @@ describe("Stub test for controller and server", () => {
...
+    };
+    return request(Server)
+      .post("/api/todo")
+      .send(todo)
+      .then((r) => {
+        expect(r.statusCode).to.equal(201);
+      });
+  });
+
+  it("should send bad request when attempt to add new todo with missing name property", () => {
```
Comment (fernandogruning) [inline backend/test/todo.controller.js]: Also, add test cases for the other missing required properties.

File: backend/server/api/controllers/todo/router.js [hunk 3]
```diff
@@ -5,3 +5,4 @@ export default express
   .Router()
   .post("/", controller.create)
   .post("/stub", controller.stub);
+//.delete("/DELETE/:id", controller.deleteByID);
```
Comment (fernandogruning) [inline backend/server/api/controllers/todo/router.js:8]: Remove this comment

File: backend/server/api/controllers/todo/controller.js [hunk 4]
```diff
@@ -1,13 +1,32 @@
...
+      description: req.body.description,
+      isDone: false,
+      dueDate: req.body.dueDate,
+    };
+    if (todo.description == undefined) {
+      //If no description was provided, then set it equal to an empty string
+      //so the data base accepts an empty description
+      todo.description = "";
+    }
+    if (todo.name == undefined) {
```
Comment (fernandogruning) [inline backend/server/api/controllers/todo/controller.js]: You're checking for `name` and `description` but what happens if the user doesn't send you `dueDate` and/or `isDone`? You should have validations for that.

File: backend/server/api/controllers/todo/controller.js [hunk 5]
```diff
@@ -1,13 +1,32 @@
-const db = require("../../services/todos.service");
+import ToDoService from "../../services/todos.service";
 
 export class Controller {
   stub(req, res) {
-    console.log(`Temporal... ${db}...`);
+    //	console.log(`Temporal... ${db}...`);
```
Comment (fernandogruning) [inline backend/server/api/controllers/todo/controller.js]: Remove comment

File: backend/server/api/controllers/todo/controller.js [hunk 6]
```diff
@@ -1,13 +1,32 @@
...
+      name: req.body.name,
+      description: req.body.description,
+      isDone: false,
+      dueDate: req.body.dueDate,
+    };
+    if (todo.description == undefined) {
+      //If no description was provided, then set it equal to an empty string
+      //so the data base accepts an empty description
+      todo.description = "";
+    }
```
Comment (fernandogruning) [inline backend/server/api/controllers/todo/controller.js]: You can refactor this by removing this block and in the `todo` object setting `description: req.body.description || ''`

**Review by fernandogruning — CHANGES_REQUESTED** [review]

**Comment by fernandogruning** [comment]
@kevinrosarioc add formatting

File: backend/server/api/controllers/todo/controller.js (hunk 6 above)
Comment (kevinrosarioc) [reply backend/server/api/controllers/todo/controller.js]: Done

File: backend/server/api/controllers/todo/controller.js (hunk 5 above)
Comment (kevinrosarioc) [reply backend/server/api/controllers/todo/controller.js]: Done

File: backend/server/api/controllers/todo/controller.js (hunk 4 above)
Comment (kevinrosarioc) [reply backend/server/api/controllers/todo/controller.js]: Done

File: backend/server/api/controllers/todo/router.js (hunk 3 above)
Comment (kevinrosarioc) [reply backend/server/api/controllers/todo/router.js]: Done

File: backend/test/todo.controller.js (hunk 2 above)
Comment (kevinrosarioc) [reply backend/test/todo.controller.js:41]: Done

File: backend/test/todo.controller.js (hunk 1 above)
Comment (kevinrosarioc) [reply backend/test/todo.controller.js:26]: Done

**Review by fernandogruning — APPROVED** [review]
